docs/configuration.md

```diff
@@ -0,0 +1,94 @@
+# Configuration
+
+See the default config [here](../config/default.yml)
+
+---
+### allowUnused
+List of modules that are allowed to be unused.
+Passed to `string.match`
+
+Please create an [issue](https://github.com/charlierudolph/dependency-lint/issues)
+anytime you need to use this
+
+Example:
+```yml
+allowUnused:
+  - mocha
+```
+
+---
+### devFilePatterns
+Files used only for development.
+Uses [minimatch](https://github.com/isaacs/minimatch)
```

Comment:
Maybe say "see [all syntax options]" with this being a link to the minimartch read me section that details the glob patterns? This way you still convey all the information (what parser you use, what the options are), with the text being less cluttered with technical implementation details.